dalm/training/generator_only/trainer.py

```diff
@@ -0,0 +1,274 @@
+import os
+from typing import Optional
+
+import torch
+from accelerate import Accelerator
+from datasets import load_dataset
+from peft import LoraConfig
+from tqdm import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer, BitsAndBytesConfig, TrainingArguments
+
+from trl import SFTTrainer
+from trl.trainer import ConstantLengthDataset
+
+import argparse
+
+
+def create_datasets(
+    dataset_name: str,
+    split: str,
+    size_valid_set: Optional[int],
+    streaming: bool,
+    shuffle_buffer: int,
+    seq_length: int,
+    num_workers: int,
+    tokenizer: AutoTokenizer,
+    formatting_func: callable,
+):
+    dataset = load_dataset(
+        dataset_name,
+        split=split,
+        num_proc=num_workers if not streaming else None,
+        streaming=streaming,
+    )
+    if streaming:
+        print("Loading the dataset in streaming mode")
```

Comment:
We should use loggers instead of raw print.  When running in cloud training environments, only the logger output is captured and the print statements get lost.

Here's an example from the DALM repo of creating a logger:

https://github.com/arcee-ai/DALM/blob/main/dalm/datasets/qa_gen/question_answer_generation.py#L14

and using it:

https://github.com/arcee-ai/DALM/blob/main/dalm/datasets/qa_gen/question_answer_generation.py#L119